tensorflow/core/ops/mkl_nn_ops.cc

```diff
@@ -803,6 +803,140 @@ REGISTER_OP("_MklDepthwiseConv2dNativeBackpropFilter")
       return Status::OK();
     });
 
+REGISTER_OP("_MklQuantizedMatMulWithBias")
+    .Input("a: T1")
+    .Input("b: T2")
+    .Input("bias: Tbias")
+    .Input("min_a: float")
+    .Input("max_a: float")
+    .Input("min_b: float")
+    .Input("max_b: float")
+    .Input("mkl_a: uint8")      // MKL second tensor
+    .Input("mkl_b: uint8")      // MKL second tensor
+    .Input("mkl_bias: uint8")   // MKL second tensor
+    .Input("mkl_min_a: uint8")  // MKL second tensor
+    .Input("mkl_max_a: uint8")  // MKL second tensor
+    .Input("mkl_min_b: uint8")  // MKL second tensor
+    .Input("mkl_max_b: uint8")  // MKL second tensor
+    .Output("out: Toutput")
+    .Output("min_out: float")
+    .Output("max_out: float")
+    .Output("mkl_out: uint8")      // MKL second tensor
+    .Output("mkl_min_out: uint8")  // MKL second tensor
+    .Output("mkl_max_out: uint8")  // MKL second tensor
+    .Attr("T1: quantizedtype")
+    .Attr("T2: quantizedtype")
+    .Attr("Tbias: {float, qint32}")
+    .Attr("T: quantizedtype")  // Additional attr "T" for MklToTf conversion
+    .Attr("Toutput: quantizedtype = DT_QINT32")
+    .Attr("transpose_a: bool = false")
+    .Attr("transpose_b: bool = false")
+    .Attr("input_quant_mode: {'MIN_FIRST', 'SCALED'} = 'MIN_FIRST'")
+    .SetShapeFn([](InferenceContext* c) {
+      TF_RETURN_IF_ERROR(shape_inference::MatMulShape(c));
+      ShapeHandle unused;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(2), 1, &unused));
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(3), 0, &unused));
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(4), 0, &unused));
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(5), 0, &unused));
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(6), 0, &unused));
+
+      c->set_output(1, c->Scalar());
```

Comment:
Please add a TODO to infer the output shape (0th output) from the shapes of `a` and `b` when available. Same for other `REGISTER_OP`. (You can make it a helper function and call it from all `_MklQuantizedMatmul*`.)

Reply:
@penpornk We have followed https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/math_ops.cc#L1658-L1666
In the above reference, I think shape is inferred in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/math_ops.cc#L1658

Reply:
You're right. Sorry about this. For some reason, I missed the line `TF_RETURN_IF_ERROR(shape_inference::MatMulShape(c));`.